src/psyclone/tests/psyir/backend/psyir_openmp_test.py

```diff
@@ -0,0 +1,239 @@
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author J. Henrichs, Bureau of Meteorology
+# -----------------------------------------------------------------------------
+
+'''Performs pytest tests on the psyclond.psyir.backend.fortran module'''
+
+from __future__ import absolute_import
+
+from psyclone.psyGen import Assignment, Reference
+from psyclone.psyir.backend.c import CWriter
+from psyclone.psyir.backend.fortran import FortranWriter
+from psyclone.tests.psyclone_test_utils import create_schedule
+
+
+# ----------------------------------------------------------------------------
+def test_nemo_omp_parallel():
+    '''Tests if an OpenMP parallel directive in NEMO is handled correctly.
+    '''
+    # Generate fparser2 parse tree from Fortran code.
+    code = '''
+        module test
+        contains
+        subroutine tmp()
+          integer :: i, a
+          integer, dimension(:) :: b
+          do i = 1, 20, 2
+            a = 2 * i
+            b(i) = b(i) + a
+          enddo
+        end subroutine tmp
+        end module test'''
+    schedule = create_schedule(code)
+    from psyclone.transformations import OMPParallelTrans
+
+    # Now apply a parallel transform
+    omp_par = OMPParallelTrans()
+    omp_par.apply(schedule[0])
+
+    fvisitor = FortranWriter()
+    result = fvisitor(schedule)
+    correct = '''!$omp parallel private(a,i)
+      do i = 1, 20, 2
+        a=2 * i
+        b(i)=b(i) + a
+      enddo
+!$omp end parallel'''
+    assert correct in result
+
+    cvisitor = CWriter()
+    result = cvisitor(schedule[0])
+    correct = '''#pragma omp parallel private(a,i)
+{
+  for(i=1-1; i<20; i+=2)
+  {
+    a = (2 * i);
+    b[i] = (b[i] + a);
+  }
+}'''
+    result = cvisitor(schedule[0])
+    assert correct in result
+
+
+# ----------------------------------------------------------------------------
+def replace_child_with_assignment(node):
+    '''Since at this stage not all node types are supported,
+    this function is used to replace the first child of the
+    given node with a simple assignment statement ('a=b').
+    This allows all tests to compare all output of the visitor
+    pattern (even though in some cases the code might not
+    compile, e.g. assignment as child of an OMP DO directive)
+
+    :param node: the node whose child is replaced.
+    :type node: :py:class:`psyclone.psyGen.Node`
+    '''
+
+    # Create a simple 'a=b' assignment statement for all tests
+    assignment = Assignment(parent=node)
+    lhs = Reference('a', assignment)
+    rhs = Reference('b', assignment)
+    assignment.addchild(lhs)
+    assignment.addchild(rhs)
+    del node.children[0]
```

Comment:
It might be more robust to simply replace `node.children[0]` with the new `assignment` object - if the `parent` has more than one child then this will produce consistent results.

Reply:
Done, though ideally I would say that addchild should take care of existing parent relationship and adjust them as well - so I would prefer addchild. But since addchild at this stage does not to anything else, I've changed it.